### PR TITLE
Ant Design 4 Form

### PR DESCRIPTION
**Pull request by rkuykendall**
**Jira:** https://thatsmighty.atlassian.net/browse/PROD-271

### CODE REVIEW

File: src/components/Form.tsx [hunk 1]
```diff
@@ -119,21 +127,15 @@ export class UnwrappedForm extends Component<IFormWrappedProps> {
...
-
-  public render() {
-    return <WrappedForm {...this.props} />;
-  }
+function Form(props: IFormProps) {
+  const [form] = AntForm.useForm();
+  return <UnwrappedForm {...props} form={form} />;
 }
 
+Form.defaultProps = {
```
Comment (ekatsuta) [inline src/components/Form.tsx:135]: we usually define `const defaultProps` at the top of the file. if we want to maintain consistency

File: src/inputs/ObjectSearchCreate.tsx [hunk 2]
```diff
@@ -69,21 +76,39 @@ class ObjectSearchCreate extends Component<IObjectSearchCreateProps> {
...
+    const createFieldsToDefaults: FieldData[] = this.createFields.map(createField => ({
+        name: createFieldAntName(fieldConfig, createField),
+        value: formManager.getDefaultValue(createField),
+      })),
+      mainFieldAsEmptyObject = { name: fieldConfig.name, value: {} };
+
+    formManager.form.setFields([mainFieldAsEmptyObject, ...createFieldsToDefaults]);
+
+    this.isAddingNew.setTrue();
+    if (onAddNewToggle) {
```
Comment (ekatsuta) [inline src/inputs/ObjectSearchCreate.tsx:96]: [nit] space above

File: src/inputs/ObjectSearchCreate.tsx [hunk 3]
```diff
@@ -69,21 +76,39 @@ class ObjectSearchCreate extends Component<IObjectSearchCreateProps> {
...
     const { onAddNewToggle, formManager, fieldConfig } = this.injected;
     this.search = search;
 
-    formManager.form.setFieldsValue({ [fieldConfig.field]: {} }, () => {
-      this.isAddingNew.setTrue();
-      if (onAddNewToggle) {
-        onAddNewToggle(true);
-      }
-    });
+    const createFieldsToDefaults: FieldData[] = this.createFields.map(createField => ({
```
Comment (ekatsuta) [inline src/inputs/ObjectSearchCreate.tsx:87]: so..`createFields` creates fields-ant fields with validation rules, label, editProps...etc., and `createFieldsToDefaults` overrides the `name` and assigns a default value to the field? Might be useful to have a short comment for each, since it's a bit hard to follow with both functions mapping over the `createFields`

File: src/inputs/ObjectSearchCreate.tsx [hunk 4]
```diff
@@ -69,21 +76,39 @@ class ObjectSearchCreate extends Component<IObjectSearchCreateProps> {
...
+      onAddNewToggle(true);
+    }
   }
 
   private async onSearch() {
-    const { onAddNewToggle, formManager, id, fieldConfig } = this.injected;
-    formManager.form.setFieldsValue({ [id]: formManager.getDefaultValue(fieldConfig) });
+    const { onAddNewToggle, formManager, fieldConfig } = this.injected;
+
+    const createFieldsToUndefined: FieldData[] = this.createFields.map(createField => ({
```
Comment (ekatsuta) [inline src/inputs/ObjectSearchCreate.tsx:104]: since a similar pattern is used above in `createFieldsToDefaults` could this map function be extracted?

File: test/features/submitButtonDisabled.test.ts [hunk 5]
```diff
@@ -32,11 +33,18 @@ describe('isSubmitButtonDisabled', () => {
...
+
       expect(isInvalid(tester)).toBe(false);
 
-      await changeInput(tester, '');
-      await tester.refresh();
+      await act(async () => {
+        await changeInput(tester, '');
+        await tester.refresh();
+      });
       expect(isInvalid(tester)).toBe(true);
```
Comment (ekatsuta) [inline test/features/submitButtonDisabled.test.ts:48]: [nit] space above

File: test/features/successText.test.ts [hunk 6]
```diff
@@ -14,6 +14,7 @@ describe('successText', () => {
     spyOn(notification, 'success');
     expect(notification.success).not.toHaveBeenCalled();
     await tester.submit();
+    await tester.refresh();
     expect(notification.success).toHaveBeenCalledWith({
```
Comment (ekatsuta) [inline test/features/successText.test.ts:18]: [nit] space above

File: test/features/successText.test.ts [hunk 7]
```diff
@@ -30,6 +31,7 @@ describe('successText', () => {
...
```
Comment (ekatsuta) [inline test/features/successText.test.ts:35]: [nit] space above

File: test/features/successText.test.ts [hunk 8]
```diff
@@ -46,6 +48,7 @@ describe('successText', () => {
     spyOn(notification, 'success');
     expect(notification.success).not.toHaveBeenCalled();
     await tester.submit();
+    await tester.refresh();
     expect(notification.success).not.toHaveBeenCalled();
```
Comment (ekatsuta) [inline test/features/successText.test.ts:52]: [nit] space above

File: src/inputs/ObjectSearchCreate.tsx (hunk 3 above)
Comment (rkuykendall) [reply src/inputs/ObjectSearchCreate.tsx:87]: Hm, yeah, I could probably use some more help figuring out what to change, but is this better?

image

**Review by ekatsuta — APPROVED** [review]
